### PR TITLE
Add trivial single-threaded prometheus exporter

### PR DESCRIPTION
**Pull request by darkk**
I decided to implement a network service instead of writing full `.prom` file on every `write_value` to a `textfile` directory polled by `node_exporter` as that needs write-and-rename code (otherwise `node_exporter` will complain about malformed file sometimes) and it'll be less trivial if `node_exporter` and `co2mond` are running under different UIDs.

### CODE REVIEW

File: co2mond/src/main.c [hunk 1]
```diff
@@ -122,33 +166,237 @@ static void
...
+
+        if (setsockopt(client_fd, SOL_SOCKET, SO_RCVTIMEO, &maxdelay, sizeof(maxdelay)) != 0 ||
+            setsockopt(client_fd, SOL_SOCKET, SO_SNDTIMEO, &maxdelay, sizeof(maxdelay)) != 0)
+        {
+            perror("setsockopt");
+            goto cleanup;
+        }
+
+        if (read_match_path(client_fd) != 0 || read_find_crlfcrlf(client_fd) != 0)
+        {
```
Comment (dmage) [inline co2mond/src/main.c:267]: @darkk what do you think about sending "400 Bad Request" in this case?

Comment (darkk) [reply co2mond/src/main.c:267]: I have weak arguments for both options (besides vague mumble about RFC-compliance as I don't send `Date` anyway).

Pro: sending `400` is a better response than RST that may aid in catching some typo somewhere.
Contra: sending `400` to a "disconnected" `client_fd` may deliver SIGPIPE in some weird case (but it should be ignored for a daemon anyway).

Thanks for highlighting the matter, it made me think of SIGPIPE that should be ignored.

I'll add `400 Bad Request` together with SIGPIPE SIG_IGN.

File: co2mond/src/main.c [hunk 2]
```diff
@@ -122,33 +167,230 @@ static void
...
+            (long long)copy.heatbeat
+        );
+flush:
+        fflush(out);
+        if (shutdown(client_fd, SHUT_WR) != 0)
+        {
+            perror("shutdown");
+            goto cleanup;
+        }
+        fgetc(out); // wait till EOF (or timeout) before calling close()
```
Comment (darkk) [inline co2mond/src/main.c:342]: TBH, I'm unsure if this `fgetc()` is actually useful as `close()` should send the queue the background unless SO_LINGER is explicitly set (according to `man 7 socket`).

**Comment by dmage** [comment]
@darkk thank you. Collecting data using Prometheus is a very common use case 👍

**Pull request merged**